src/nacl/public.py

```diff
@@ -92,17 +92,18 @@ def __init__(self, private_key, encoder=encoding.RawEncoder):
 
         self._private_key = private_key
         self.public_key = PublicKey(raw_public_key)
+        self._hashbytes = b'p:' + bytes(self.public_key)
 
     def __bytes__(self):
         return self._private_key
 
     def __hash__(self):
-        return hash(bytes(self))
+        return hash(self._hashbytes)
```

Comment:
they can still end up in the same bucket, the hashes don't need to collide, they just need to collide mod len(table). I don't think the `__eq__` gets involved then though, since the hashes are checked for equality first.

I'm still not sure I understand _why_ this change though.